### PR TITLE
Firebase Authentication 

### PR DESCRIPTION
**Pull request by dmroberts302**
Added firebase authentication to the codebase. When you navigate to the /Login route you have the option to Login or Register, if you are successful at either logging in or registering you will be redirected to the Homepage where your JWT will be logged to the console, if you are not successful you will not be redirected and you can try again.

### CODE REVIEW

**Comment by dmroberts302** [comment]
Would you also like me to place the firebase credentials in a .env file? I was not sure how you wanted me to handle the firebase credentials.

**Comment by a-mufasa** [comment]
> Would you also like me to place the firebase credentials in a .env file? I was not sure how you wanted me to handle the firebase credentials.

I'd prefer having it in a .env file. Not sure if we have the dotenv package tho so it's fine to leave it like this for now then we can set up a .env file for everything later down the line. If we are actually wanting to be secure, we should work with @JackTNorris to get secrets set up.

File: src/screens/HomeScreen.tsx [hunk 1]
```diff
@@ -19,6 +32,10 @@ const HomeScreen: FunctionComponent<HomeScreenProps> = (
             <Link to="/checkInOut" className="underline text-blue-600 ">
                 Check In Out
             </Link>
+            <br></br>
+            <Link to="/Login" className="underline text-blue-600 ">
```
Comment (a-mufasa) [inline src/screens/HomeScreen.tsx:36]: we should probably maintain snake case for our routes: `/login`

File: src/screens/Login.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+import { Button } from '@mui/material';
+
+const Login = () => {
+    const navigate = useNavigate();
+    const [email, setEmail] = useState('');
+    const [password, setPassword] = useState('');
+
+    const onLogin = (e: any) => {
+        e.preventDefault();
+        signInWithEmailAndPassword(auth, email, password)
```
Comment (a-mufasa) [inline src/screens/Login.tsx:17]: not super familiar with firebases auth functions but is there a reason why register is async and login isnt?

**Review by a-mufasa — APPROVED** [review]
i left some suggestions but not sure if we wanna get this merged asap or if yall will have time to review this so ill give it an approval in case.

**Comment by JackTNorris** [comment]
Adding to what Ahmed said, it's alright for now if you don't put the credentials in a .env file. We'll just have to regen credentials and whatnot when we make the site public. This looks good for now

File: src/screens/Login.tsx (hunk 2 above)
Comment (dmroberts302) [reply src/screens/Login.tsx:17]: sorry I forgot to remove the async/await from that function!

**Pull request merged**